internal/ingest/ingest.go

```diff
@@ -799,14 +809,14 @@ func (ing *Ingester) ingestWorker() {
 			pc := ing.providersBeingProcessed[pid]
 			ing.providersBeingProcessedMu.Unlock()
 			pc <- struct{}{}
-			ing.ingestWorkerLogic(pid)
-			ing.handlePendingAnnounce(pid)
+			ing.ingestWorkerLogic(ctx, pid)
```

Comment:
Should the `select` above also select on `ctx.Done()`?

Reply:
I think that this cannot exit while items are still being written to `ing.toWorkers`, otherwise those things will block on writing the channel.